In [107]:
import pandas as pd
import numpy as np
import os


ROOT = "c:\\Users\\1004c\\Desktop\\통계청 논문 공모전"
ROOT

'c:\\Users\\1004c\\Desktop\\통계청 논문 공모전'

## Variable Selection

In [108]:
raw = pd.read_excel(os.path.join(ROOT, "Data", "GP19__2020.xlsx"))

In [109]:
raw.columns

Index(['g191pid', 'g191branch', 'g191found', 'g191course', 'g191daynight',
       'g191majorcat', 'g191dpmt', 'g191dpmt_n', 'g191major', 'g191major_n',
       ...
       'g191r021', 'g191r022', 'g191r023', 'g191r024', 'g191r025', 'g191r026',
       'g191r028', 'g191r029', 'g191r027', 'g191wt'],
      dtype='object', length=1332)

- 개인 및 가족: X1
- 학교 특성: X2
- 취업 준비 활동: X3
---
- 취업 관련: Y
- 정책 변수(Treatment): T

In [110]:
variable_mapping = {
	"g191found": "X2 - 국공사립",
	"g191majorcat": "X2 - 전공계열",
	"g191school": "X2 - 학교유형",
	"g191province": "X2 - 학교소재권역",
	
    "g191sex": "X1 - 성별",
	"g191age": "X1 - 연령",
	
    "g191d001": "Y - 대학 졸업 후 일자리를 가진 경험의 유무",
	"g191d002": "Y - 첫 일자리 시작시기_년",
	"g191d026": "Y - <임금근로자> 첫 일자리",
	"g191d080": "Y - <비임금근로자> 첫 일자리",
	"g191d130": "Y - 첫 직장(일자리)에 대한 전반적인 만족도",
	
    "g191f034": "X3 - 진로선택 및 취업준비 프로그램 1- 진로, 취업 관련 교과목_참여 여부",
	"g191f036": "X3 - 진로선택 및 취업준비 프로그램 2- 직장체험프로그램(인턴 포함)_참여 여부",
	"g191f037": "X3 - 진로선택 및 취업준비 프로그램 3- 인적성 검사 등 직업심리검사_참여 여부",
	"g191f038": "X3 - 진로선택 및 취업준비 프로그램 4- 교내 취업박람회_참여 여부",
	"g191f039": "X3 - 진로선택 및 취업준비 프로그램 5- 진로 관련 개인 및 집단상담 프로그램_참여 여부",
	"g191f040": "X3 - 진로선택 및 취업준비 프로그램 6- 면접기술, 이력서 작성 프로그램_참여 여부",
	"g191f041": "X3 - 진로선택 및 취업준비 프로그램 7- 취업캠프_참여 여부",
	"g191f042": "X3 - 진로선택 및 취업준비 프로그램 8- 기업채용설명회_참여 여부",
	"g191f043": "X3 - 진로선택 및 취업준비 프로그램 9- 기타_참여 여부",
	
    "g191f074": "X1 - 졸업 평점 점수",
	"g191h001": "X3 - 대학 재학 중 일자리 경험의 유무",
	"g191i002": "X3 - 어학연수 경험 횟수",
	"g191m002": "X3 - 자격증 소지 개수",
	
    "g191k004": "X3 - 취업 준비활동 경험여부 1- 기업체 직무적성검사(SSAT 등) 공부",
	"g191k005": "X3 - 취업 준비활동 경험여부 2-영어 등 외국어 공부",
	"g191k006": "X3 - 취업 준비활동 경험여부 3- 봉사활동",
	"g191k007": "X3 - 취업 준비활동 경험여부 4-공모전 수상",
	"g191k008": "X3 - 취업 준비활동 경험여부 5-자격증 준비",
	"g191k009": "X3 - 취업 준비활동 경험여부 6-대외활동(체험단, 마케터, 서포터즈 등)",
	"g191k010": "X3 - 취업 준비활동 경험여부 7-외모관리(다이어트, 시술, 수술 등)",
	"g191k011": "X3 - 취업 준비활동 경험여부 8-이력서 작성 및 면접훈련 교육",
	"g191k012": "X3 - 취업 준비활동 경험여부 9-(모의)면접훈련",
	
	"g191o035": "T1 - 중소기업 청년취업인턴제_들어본 경험",
    "g191o037": "T1 - 중소기업 청년취업인턴제_참여경험",
	"g191o044": "T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_들어본 경험",
	"g191o046": "T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_참여경험",
	"g191o106": "T3 - 취업성공패키지_청년고용정책 인지여부",
	"g191o108": "T3 - 취업성공패키지_청년고용정책 참여 여부",
	"g191o097": "T4 - 내일배움카드(구직업 능력 개발 계좌제)_들어본 경험",
	"g191o099": "T4 - 내일배움카드(구직업 능력 개발 계좌제)_참여경험",
	
    "g191p001": "X1 - 혼인 여부",
	"g191p026": "X1 - 아버님의 최종 학력",
	"g191p029": "X1 - 어머님의 최종 학력",
	"g191p034": "X1 - 대학 입학(편입) 당시 부모님의 월평균 소득",
	
    "g191r001": "Y - 전년도 소득여부",
	"g191r002": "Y - 전년도 근로/사업 소득액"
}

In [111]:
df_selected = raw[list(variable_mapping.keys())]
df_selected.rename(columns=variable_mapping, inplace=True)
print(df_selected.head())

   X2 - 국공사립  X2 - 전공계열  X2 - 학교유형  X2 - 학교소재권역  X1 - 성별    X1 - 연령  \
0          3          1          1            1        2  26.333333   
1          3          1          1            1        2  22.333333   
2          3          1          1            2        2  21.666667   
3          3          1          1            2        1  25.916667   
4          3          1          1            4        1  25.500000   

   Y - 대학 졸업 후 일자리를 가진 경험의 유무  Y - 첫 일자리 시작시기_년  Y - <임금근로자> 첫 일자리  \
0                         1.0            2019.0                2.0   
1                         2.0               NaN                NaN   
2                         1.0            2019.0                2.0   
3                         2.0               NaN                NaN   
4                         NaN               NaN                NaN   

   Y - <비임금근로자> 첫 일자리  ...  T3 - 취업성공패키지_청년고용정책 인지여부  \
0                 NaN  ...                         2   
1                 NaN  ...              

C:\Users\1004c\AppData\Local\Temp\ipykernel_12868\462389319.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected.rename(columns=variable_mapping, inplace=True)


In [112]:
df_selected.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18271 entries, 0 to 18270
Data columns (total 47 columns):
 #   Column                                               Non-Null Count  Dtype  
---  ------                                               --------------  -----  
 0   X2 - 국공사립                                            18271 non-null  int64  
 1   X2 - 전공계열                                            18271 non-null  int64  
 2   X2 - 학교유형                                            18271 non-null  int64  
 3   X2 - 학교소재권역                                          18271 non-null  int64  
 4   X1 - 성별                                              18271 non-null  int64  
 5   X1 - 연령                                              18271 non-null  float64
 6   Y - 대학 졸업 후 일자리를 가진 경험의 유무                           9118 non-null   float64
 7   Y - 첫 일자리 시작시기_년                                     5614 non-null   float64
 8   Y - <임금근로자> 첫 일자리                                    5567 non-null

In [113]:
df_selected["Y - 대학 졸업 후 일자리를 가진 경험의 유무"].value_counts()

Y - 대학 졸업 후 일자리를 가진 경험의 유무
1.0    5614
2.0    3504
Name: count, dtype: int64

- <임금 근로자>: 5567
- <비임금 근로자>: 37
- 취업 X: 12657

## X1 - 개인 및 가족

- 성별: 1(남자), 2(여자)
- 연령: 만 20-34세로 제한
- 졸업 평점 점수
- 혼인 여부: 1(미혼), 2(기혼), 3(이혼), 4(사별)
- 부모님의 최종 학력: 평균(6: 초졸, 9: 중졸, 12: 고졸, 14: 전문대졸, 16: 대졸, 18: 대학원졸)
- 부모님 월평균 소득

In [114]:
x1_columns = [col for col in df_selected.columns if col.startswith("X1")]
df_selected[x1_columns].columns

Index(['X1 - 성별', 'X1 - 연령', 'X1 - 졸업 평점 점수', 'X1 - 혼인 여부', 'X1 - 아버님의 최종 학력',
       'X1 - 어머님의 최종 학력', 'X1 - 대학 입학(편입) 당시 부모님의 월평균 소득'],
      dtype='object')

### 성별
- 남자(0), 여자(1)로 변환

In [115]:
df_selected["X1 - 성별"].value_counts()

X1 - 성별
2    9481
1    8790
Name: count, dtype: int64

In [116]:
df_selected.loc[:, "X1 - 성별"] = df_selected["X1 - 성별"].map({1: 0, 2: 1})

### 연령
- 만 20-34세로 filtering
- 더미 변수로: 20-24세 / 25-29세 / 30-34세

In [117]:
df_selected = df_selected[(df_selected["X1 - 연령"] >= 20) & (df_selected["X1 - 연령"] <= 34)]
df_selected["X1 - 연령"].value_counts()

X1 - 연령
26.500000    340
26.333333    332
25.666667    330
24.666667    326
26.416667    323
            ... 
21.166667      1
21.416667      1
20.833333      1
33.583333      1
20.750000      1
Name: count, Length: 156, dtype: int64

In [118]:
# 1단계: 구간화 함수 정의
def categorize_age(age):
    if 20 <= age <= 24:
        return "20_24"
    elif 25 <= age <= 29:
        return "25_29"
    elif 30 <= age <= 34:
        return "30_34"
    else:
        return np.nan

# 2단계: 연령대 범주 변수 생성
df_selected["X1 - 연령대"] = df_selected["X1 - 연령"].apply(categorize_age)

# 3단계: 더미 변수 생성 (원하는 경우 drop_first=True 가능)
# df_selected = pd.get_dummies(df_selected, columns=["연령대"], prefix="X1 - 연령", dtype=int)

# 연령대를 범주형(factor) 변수로 지정 - ordinal
df_selected["X1 - 연령대"] = pd.Categorical(
    df_selected["X1 - 연령대"],
    categories=["20_24", "25_29", "30_34"],
    ordered=True
)

In [119]:
df_selected["X1 - 연령대"].value_counts()

X1 - 연령대
25_29    10775
20_24     3025
30_34      519
Name: count, dtype: int64

### 졸업 평점 점수
- no change

In [120]:
df_selected["X1 - 졸업 평점 점수"].agg(["min", "max"])

min    1.0
max    4.5
Name: X1 - 졸업 평점 점수, dtype: float64

In [121]:
df_selected["X1 - 졸업 평점 점수"].isna().sum()

1294

In [122]:
df_selected["X1 - 졸업 평점 결측"] = df_selected["X1 - 졸업 평점 점수"].isna().astype(int)
df_selected["X1 - 졸업 평점 점수"] = df_selected["X1 - 졸업 평점 점수"].fillna(
    df_selected["X1 - 졸업 평점 점수"].mean()
)

### 혼인 여부
- 이혼(3)만 1, 나머지는 0

In [123]:
df_selected["X1 - 혼인 여부"].value_counts()

X1 - 혼인 여부
1    17609
2      283
3       10
4        6
Name: count, dtype: int64

In [124]:
df_selected["X1 - 혼인 여부"] = (df_selected["X1 - 혼인 여부"] == 3).astype(int)

### 부모님의 최종 학력

- -1(모름/무응답), 1(무학), 2(초등학교), 3(중학교), 4(고등학교), 5(2-3년제 대학(초급대)), 6(4년제 대학), 7(대학원) 

In [125]:
df_selected["X1 - 아버님의 최종 학력"].value_counts()
df_selected["X1 - 어머님의 최종 학력"].value_counts()

X1 - 어머님의 최종 학력
 4    8485
 6    4404
 5    1766
-1    1564
 7     720
 3     619
 2     312
 1      38
Name: count, dtype: int64

In [126]:
# 학력 코드 → 연한 매핑
edu_year_map = {
    -1: np.nan,  # 모름/무응답
    1: 0,
    2: 6,
    3: 9,
    4: 12,
    5: 14,
    6: 16,
    7: 18
}

# 각 부모의 학력을 연한으로 매핑
father_years = df_selected["X1 - 아버님의 최종 학력"].map(edu_year_map)
mother_years = df_selected["X1 - 어머님의 최종 학력"].map(edu_year_map)

# 평균 학력 연한 계산 (NaN 제외한 평균)
df_selected["X1 - 부모학력"] = pd.concat([father_years, mother_years], axis=1).mean(axis=1, skipna=True)

### 부모님 월평균 소득
- 1(100만원 미만)
- 2(100~200만원 미만)
- 3(200~300만원 미만)
- 4(300~400만원 미만)
- 5(400~500만원 미만)
- 6(500~700만원 미만)
- 7(700~1,000만원 미만)
- 8(1,000만원 이상)
- 9(소득없음)
- 10(안계심(사망))

- 더미변수로: 300만원 미만 / 300-500만원 / 500만원 이상

In [127]:
# 단계 1: 구간 분류 함수 정의
def categorize_income(code):
    if code in [1, 2, 3]:
        return "저소득"
    elif code in [4, 5]:
        return "중소득"
    elif code in [6, 7, 8]:
        return "고소득"
    else:
        return np.nan  # 9 (소득없음), 10 (사망) → 결측 처리

# 단계 2: 새 변수 생성
df_selected["X1 - 부모소득구간"] = df_selected["X1 - 대학 입학(편입) 당시 부모님의 월평균 소득"].apply(categorize_income)

# 단계 3: 더미 변수로 변환 (원한다면 drop_first=True로 기준 더미 제거 가능)
# df_selected = pd.get_dummies(df_selected, columns=["부모소득구간"], prefix="X1 - 부모소득", dtype=int)

# Categorical(ordinal) 변수 생성
df_selected["X1 - 부모소득구간"] = pd.Categorical(
    df_selected["X1 - 부모소득구간"],
    categories=["저소득", "중소득", "고소득"],
    ordered=True  # 순서형으로 하고 싶지 않다면 False로
)

In [128]:
df_selected["X1 - 부모소득구간"].value_counts()

X1 - 부모소득구간
중소득    4990
고소득    4084
저소득    2974
Name: count, dtype: int64

## X2 - 학교 특성

- 국공사립: 1(국립), 2(공립), 3(사립)
- 전공계열: 1(인문), 2(사회), 3(교육), 4(공학), 5(자연), 6(의약), 7(예체능)
- 학교유형: 1(2-3년제), 2(4년제), 3(교육대)
- 학교소재권역: 1(서울권), 2(경기원), 3(충청권), 4(경상권), 5(전라권)

In [129]:
x2_columns = [col for col in df_selected.columns if col.startswith("X2")]
df_selected[x2_columns]

X2 - 국공사립  X2 - 전공계열  X2 - 학교유형  X2 - 학교소재권역
0              3          1          1            1
1              3          1          1            1
2              3          1          1            2
3              3          1          1            2
4              3          1          1            4
...          ...        ...        ...          ...
18266          3          7          2            4
18267          3          7          2            5
18268          3          3          1            4
18269          3          3          1            2
18270          3          3          1            2

[17908 rows x 4 columns]

### 설립 유형 및 소재지

In [130]:
# 설립유형: 국공립(1 or 2)이면 1, 사립(3)이면 0
df_selected["X2 - 설립유형_국공립"] = df_selected["X2 - 국공사립"].isin([1, 2]).astype(int)

In [131]:
df_selected["X2 - 설립유형_국공립"].value_counts()

X2 - 설립유형_국공립
0    13795
1     4113
Name: count, dtype: int64

### 소재지

In [132]:
# 소재지: 서울권(1)이면 0, 지방권(2~5)이면 1
df_selected["X2 - 소재지_지방"] = (df_selected["X2 - 학교소재권역"] != 1).astype(int)

In [133]:
df_selected["X2 - 소재지_지방"].value_counts()

X2 - 소재지_지방
1    13831
0     4077
Name: count, dtype: int64

### 전공계열
- 1(인문), 2(사회), 3(교육), 4(공학), 5(자연), 6(의약), 7(예체능)
- 사회(1-3)/공학(4)/자연(5-6)/예체능(7)로 구분

In [134]:
df_selected["X2 - 전공계열"].value_counts()

X2 - 전공계열
4    4956
2    3526
5    2440
1    2347
7    1985
3    1410
6    1244
Name: count, dtype: int64

In [135]:
# 그룹 매핑 함수
def map_major_group(code):
    if code in [1, 2, 3]:
        return "사회계열"
    elif code in [5, 6]:
        return "자연계열"
    elif code == 4:
        return "공학계열"
    elif code == 7:
        return "예체능계열"

# 그룹 변수 생성
df_selected["X2 - 전공그룹"] = df_selected["X2 - 전공계열"].apply(map_major_group)

# 더미 변수 생성 (drop_first는 기준계열 지정시 사용 가능)
# df_selected = pd.get_dummies(df_selected, columns=["X2 - 전공그룹"])

# 범주형 변수로 지정 - 순서형 X, 명목형
df_selected["X2 - 전공그룹"] = pd.Categorical(
    df_selected["X2 - 전공그룹"],
    categories=["사회계열", "자연계열", "공학계열", "예체능계열"],  # 명시적 순서 없이 나열
    ordered=False  # 순서형 아님 (명목형)
)

In [136]:
df_selected["X2 - 전공그룹"].value_counts()

X2 - 전공그룹
사회계열     7283
공학계열     4956
자연계열     3684
예체능계열    1985
Name: count, dtype: int64

### 학교 유형(4년제, 전문대)
- 1(2-3년제), 2(4년제), 3(교육대)

In [137]:
df_selected["X2 - 학교유형"].value_counts()

X2 - 학교유형
2    13841
1     3772
3      295
Name: count, dtype: int64

In [138]:
df_selected["X2 - 학교유형_4년제이상"] = df_selected["X2 - 학교유형"].isin([2, 3]).astype(int)

## X3 - 취업 준비

- 진로선택 및 취업준비 프로그램: 모아서 점수내기
- 대학 재학 중 일자리 경험의 유무
- 어학연수 경험 횟수
- 자격증 소지 개수
- 취업 준비활동 경험여부: 모아서 점수 내기

In [139]:
x3_columns = [col for col in df_selected.columns if col.startswith("X3")]
df_selected[x3_columns].columns

Index(['X3 - 진로선택 및 취업준비 프로그램 1- 진로, 취업 관련 교과목_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 2- 직장체험프로그램(인턴 포함)_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 3- 인적성 검사 등 직업심리검사_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 4- 교내 취업박람회_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 5- 진로 관련 개인 및 집단상담 프로그램_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 6- 면접기술, 이력서 작성 프로그램_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 7- 취업캠프_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 8- 기업채용설명회_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 9- 기타_참여 여부', 'X3 - 대학 재학 중 일자리 경험의 유무',
       'X3 - 어학연수 경험 횟수', 'X3 - 자격증 소지 개수',
       'X3 - 취업 준비활동 경험여부 1- 기업체 직무적성검사(SSAT 등) 공부',
       'X3 - 취업 준비활동 경험여부 2-영어 등 외국어 공부', 'X3 - 취업 준비활동 경험여부 3- 봉사활동',
       'X3 - 취업 준비활동 경험여부 4-공모전 수상', 'X3 - 취업 준비활동 경험여부 5-자격증 준비',
       'X3 - 취업 준비활동 경험여부 6-대외활동(체험단, 마케터, 서포터즈 등)',
       'X3 - 취업 준비활동 경험여부 7-외모관리(다이어트, 시술, 수술 등)',
       'X3 - 취업 준비활동 경험여부 8-이력서 작성 및 면접훈련 교육', 'X3 - 취업 준비활동 경험여부 9-(모의)면접훈련'],
      dtype='object')

### 진로선택 및 취업준비 프로그램

In [140]:
participation_cols = [
    'X3 - 진로선택 및 취업준비 프로그램 1- 진로, 취업 관련 교과목_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 2- 직장체험프로그램(인턴 포함)_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 3- 인적성 검사 등 직업심리검사_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 4- 교내 취업박람회_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 5- 진로 관련 개인 및 집단상담 프로그램_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 6- 면접기술, 이력서 작성 프로그램_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 7- 취업캠프_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 8- 기업채용설명회_참여 여부',
    'X3 - 진로선택 및 취업준비 프로그램 9- 기타_참여 여부'
]

df_selected["X3 - 진로교육_참여개수"] = df_selected[participation_cols].replace({2: 0}).sum(axis=1)

In [141]:
df_selected[participation_cols].info()

<class 'pandas.core.frame.DataFrame'>
Index: 17908 entries, 0 to 18270
Data columns (total 9 columns):
 #   Column                                               Non-Null Count  Dtype
---  ------                                               --------------  -----
 0   X3 - 진로선택 및 취업준비 프로그램 1- 진로, 취업 관련 교과목_참여 여부         17908 non-null  int64
 1   X3 - 진로선택 및 취업준비 프로그램 2- 직장체험프로그램(인턴 포함)_참여 여부       17908 non-null  int64
 2   X3 - 진로선택 및 취업준비 프로그램 3- 인적성 검사 등 직업심리검사_참여 여부       17908 non-null  int64
 3   X3 - 진로선택 및 취업준비 프로그램 4- 교내 취업박람회_참여 여부              17908 non-null  int64
 4   X3 - 진로선택 및 취업준비 프로그램 5- 진로 관련 개인 및 집단상담 프로그램_참여 여부  17908 non-null  int64
 5   X3 - 진로선택 및 취업준비 프로그램 6- 면접기술, 이력서 작성 프로그램_참여 여부     17908 non-null  int64
 6   X3 - 진로선택 및 취업준비 프로그램 7- 취업캠프_참여 여부                  17908 non-null  int64
 7   X3 - 진로선택 및 취업준비 프로그램 8- 기업채용설명회_참여 여부               17908 non-null  int64
 8   X3 - 진로선택 및 취업준비 프로그램 9- 기타_참여 여부                    17908 non-null  int64
dtypes: int64(9)


In [142]:
df_selected["X3 - 진로교육_참여개수"].dtype

dtype('int64')

### 취업 준비활동 경험여부

In [143]:
# 취업 준비활동 변수 목록
prep_activity_cols = [
    'X3 - 취업 준비활동 경험여부 1- 기업체 직무적성검사(SSAT 등) 공부',
    'X3 - 취업 준비활동 경험여부 2-영어 등 외국어 공부',
    'X3 - 취업 준비활동 경험여부 3- 봉사활동',
    'X3 - 취업 준비활동 경험여부 4-공모전 수상',
    'X3 - 취업 준비활동 경험여부 5-자격증 준비',
    'X3 - 취업 준비활동 경험여부 6-대외활동(체험단, 마케터, 서포터즈 등)',
    'X3 - 취업 준비활동 경험여부 7-외모관리(다이어트, 시술, 수술 등)',
    'X3 - 취업 준비활동 경험여부 8-이력서 작성 및 면접훈련 교육',
    'X3 - 취업 준비활동 경험여부 9-(모의)면접훈련'
]

# 참여 개수 변수 생성
df_selected["X3 - 취업준비활동_참여개수"] = df_selected[prep_activity_cols].fillna(0).replace({2: 0}).sum(axis=1)
df_selected["X3 - 취업준비활동_참여개수"] = df_selected["X3 - 취업준비활동_참여개수"].astype(int)

In [144]:
df_selected[prep_activity_cols].info()

<class 'pandas.core.frame.DataFrame'>
Index: 17908 entries, 0 to 18270
Data columns (total 9 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   X3 - 취업 준비활동 경험여부 1- 기업체 직무적성검사(SSAT 등) 공부  5866 non-null   float64
 1   X3 - 취업 준비활동 경험여부 2-영어 등 외국어 공부             5866 non-null   float64
 2   X3 - 취업 준비활동 경험여부 3- 봉사활동                   5866 non-null   float64
 3   X3 - 취업 준비활동 경험여부 4-공모전 수상                  5866 non-null   float64
 4   X3 - 취업 준비활동 경험여부 5-자격증 준비                  5866 non-null   float64
 5   X3 - 취업 준비활동 경험여부 6-대외활동(체험단, 마케터, 서포터즈 등)  5866 non-null   float64
 6   X3 - 취업 준비활동 경험여부 7-외모관리(다이어트, 시술, 수술 등)    5866 non-null   float64
 7   X3 - 취업 준비활동 경험여부 8-이력서 작성 및 면접훈련 교육        5866 non-null   float64
 8   X3 - 취업 준비활동 경험여부 9-(모의)면접훈련                5866 non-null   float64
dtypes: float64(9)
memory usage: 1.4 MB


In [145]:
df_selected["X3 - 취업준비활동_참여개수"].value_counts()

X3 - 취업준비활동_참여개수
0    12116
3     1154
2     1127
4     1042
5      810
1      765
6      472
7      245
8      121
9       56
Name: count, dtype: int64

###  대학 재학 중 일자리 경험의 유무
- 1: 있다, 2: 없다

In [173]:
df_selected["X3 - 대학 재학 중 일자리 경험의 유무"].value_counts()

X3 - 대학 재학 중 일자리 경험의 유무
1    9991
0    7917
Name: count, dtype: int64

In [147]:
df_selected["X3 - 대학 재학 중 일자리 경험의 유무"] = df_selected["X3 - 대학 재학 중 일자리 경험의 유무"].replace({2: 0})

### 어학연수 경험 횟수

In [148]:
df_selected["X3 - 어학연수 경험 횟수"].value_counts()

X3 - 어학연수 경험 횟수
1.0    1463
2.0     184
3.0      22
4.0       5
5.0       1
Name: count, dtype: int64

In [149]:
df_selected["X3 - 어학연수 경험 횟수"] = df_selected["X3 - 어학연수 경험 횟수"].fillna(0).astype(int)

### 자격증 소지 개수

In [150]:
df_selected["X3 - 자격증 소지 개수"].agg(["min", "max"])

min     1.0
max    16.0
Name: X3 - 자격증 소지 개수, dtype: float64

In [151]:
df_selected["X3 - 자격증 소지 개수"] = df_selected["X3 - 자격증 소지 개수"].fillna(0).astype(int)

In [152]:
df_selected["X3 - 자격증 소지 개수"].value_counts()

X3 - 자격증 소지 개수
0     8077
1     4978
2     2560
3     1184
4      540
5      272
6      141
7       60
8       43
10      20
9       14
12       6
11       5
13       5
16       1
14       1
15       1
Name: count, dtype: int64

## Y - 취업 여부, 만족도

- 대학 졸업 후 일자리를 가진 경험의 유무: 1(예), 2(아니오)
- 첫 직장(일자리)에 대한 전반적인 만족도: 1(매우 불만족), 2(불만족), 3(보통), 4(만족), 5(매우 만족)

In [153]:
y_columns = [col for col in df_selected.columns if col.startswith("Y")]
df_selected[y_columns].columns

Index(['Y - 대학 졸업 후 일자리를 가진 경험의 유무', 'Y - 첫 일자리 시작시기_년', 'Y - <임금근로자> 첫 일자리',
       'Y - <비임금근로자> 첫 일자리', 'Y - 첫 직장(일자리)에 대한 전반적인 만족도', 'Y - 전년도 소득여부',
       'Y - 전년도 근로/사업 소득액'],
      dtype='object')

### 대학 졸업 후 일자리를 가진 경험의 유무
- 1(예), 2(아니오)

In [174]:
df_selected["Y - 대학 졸업 후 일자리를 가진 경험의 유무"].value_counts()

Y - 대학 졸업 후 일자리를 가진 경험의 유무
0    12351
1     5557
Name: count, dtype: int64

In [155]:
df_selected["Y - 대학 졸업 후 일자리를 가진 경험의 유무"] = df_selected["Y - 대학 졸업 후 일자리를 가진 경험의 유무"].replace({2: 0, 1: 1}).fillna(0).astype(int)

### 임금근로자
- one-hot encoding
- 1: if 값이 있음, 0: else
- 질문: <임금근로자> 첫 일자리에서 근로계약기간을 별도로 정했는지 여부
- 1(정해져 있었다), 2(정해져 있지 않았다)

In [175]:
df_selected["Y - <임금근로자> 첫 일자리"].value_counts()

Y - <임금근로자> 첫 일자리
0    12395
1     5513
Name: count, dtype: int64

In [157]:
df_selected["Y - <임금근로자> 첫 일자리"] = df_selected["Y - <임금근로자> 첫 일자리"].notna().astype(int)

### 비임금근로자
- one-hot encoding
- 1: if 값이 있음, 0: else

- 질문: <비임금근로자> 첫 일자리의 종류
- 1(혼자 창업), 2(동업자와 함께 창업), 3(기존에 운영되던 것을 인수), 4(기존 사업에 동업으로 참여(일정지분보유형태)), 5(가족, 친지가 하던 사업을 물려받음), 6(사업자 등록없이 혼자 일함(과외, 레슨 등))

In [176]:
df_selected["Y - <비임금근로자> 첫 일자리"].value_counts()

Y - <비임금근로자> 첫 일자리
0    17873
1       35
Name: count, dtype: int64

In [159]:
df_selected["Y - <비임금근로자> 첫 일자리"] = df_selected["Y - <비임금근로자> 첫 일자리"].notna().astype(int)

### 첫 직장(일자리)에 대한 전반적인 만족도

In [160]:
df_selected["Y - 첫 직장(일자리)에 대한 전반적인 만족도"].value_counts()

Y - 첫 직장(일자리)에 대한 전반적인 만족도
3.0    1980
4.0    1472
2.0    1092
1.0     584
5.0     429
Name: count, dtype: int64

In [161]:
df_selected["Y - 첫 직장(일자리)에 대한 전반적인 만족도"] = \
    df_selected["Y - 첫 직장(일자리)에 대한 전반적인 만족도"].fillna(0).astype(int)

### 전년도 근로/사업 소득액

In [162]:
df_selected["Y - 전년도 근로/사업 소득액"].info()

<class 'pandas.core.series.Series'>
Index: 17908 entries, 0 to 18270
Series name: Y - 전년도 근로/사업 소득액
Non-Null Count  Dtype  
--------------  -----  
12639 non-null  float64
dtypes: float64(1)
memory usage: 279.8 KB


In [163]:
df_selected["Y - 전년도 근로/사업 소득액"] = df_selected["Y - 전년도 근로/사업 소득액"].fillna(0)

In [164]:
df_selected["Y - log 소득"] = np.log1p(df_selected["Y - 전년도 근로/사업 소득액"].fillna(0).clip(lower=0))

### 전년도 소득여부

In [177]:
df_selected["Y - 전년도 소득여부"].value_counts()

Y - 전년도 소득여부
1    12639
2     5269
Name: count, dtype: int64

## T - Treatment
- "들어본_경험"이 없는 집단은 제외
- 1(있다), 2(없다)

In [166]:
t_columns = [col for col in df_selected.columns if col.startswith("T")]
df_selected[t_columns].columns

Index(['T1 - 중소기업 청년취업인턴제_들어본 경험', 'T1 - 중소기업 청년취업인턴제_참여경험',
       'T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_들어본 경험',
       'T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_참여경험', 'T3 - 취업성공패키지_청년고용정책 인지여부',
       'T3 - 취업성공패키지_청년고용정책 참여 여부', 'T4 - 내일배움카드(구직업 능력 개발 계좌제)_들어본 경험',
       'T4 - 내일배움카드(구직업 능력 개발 계좌제)_참여경험'],
      dtype='object')

In [167]:
policy_cols = [
    'T1 - 중소기업 청년취업인턴제_들어본 경험',
    'T1 - 중소기업 청년취업인턴제_참여경험',
    'T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_들어본 경험',
    'T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_참여경험',
    'T3 - 취업성공패키지_청년고용정책 인지여부',
    'T3 - 취업성공패키지_청년고용정책 참여 여부',
    'T4 - 내일배움카드(구직업 능력 개발 계좌제)_들어본 경험',
    'T4 - 내일배움카드(구직업 능력 개발 계좌제)_참여경험'
]

# 1: 있다 → 1, 2: 없다 → 0
df_selected[policy_cols] = df_selected[policy_cols].replace({2: 0, 1: 1})

# 참여경험 변수만 따로 선택하여 NaN → 0
participation_cols = [col for col in policy_cols if "참여경험" in col or "참여 여부" in col]
df_selected[participation_cols] = df_selected[participation_cols].fillna(0).astype(int)

In [168]:
df_selected[policy_cols].info()

<class 'pandas.core.frame.DataFrame'>
Index: 17908 entries, 0 to 18270
Data columns (total 8 columns):
 #   Column                                 Non-Null Count  Dtype
---  ------                                 --------------  -----
 0   T1 - 중소기업 청년취업인턴제_들어본 경험               17908 non-null  int64
 1   T1 - 중소기업 청년취업인턴제_참여경험                 17908 non-null  int32
 2   T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_들어본 경험  17908 non-null  int64
 3   T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_참여경험    17908 non-null  int32
 4   T3 - 취업성공패키지_청년고용정책 인지여부               17908 non-null  int64
 5   T3 - 취업성공패키지_청년고용정책 참여 여부              17908 non-null  int32
 6   T4 - 내일배움카드(구직업 능력 개발 계좌제)_들어본 경험      17908 non-null  int64
 7   T4 - 내일배움카드(구직업 능력 개발 계좌제)_참여경험        17908 non-null  int32
dtypes: int32(4), int64(4)
memory usage: 979.3 KB


## Final

In [182]:
df_selected.columns

Index(['X2 - 국공사립', 'X2 - 전공계열', 'X2 - 학교유형', 'X2 - 학교소재권역', 'X1 - 성별',
       'X1 - 연령', 'Y - 대학 졸업 후 일자리를 가진 경험의 유무', 'Y - 첫 일자리 시작시기_년',
       'Y - <임금근로자> 첫 일자리', 'Y - <비임금근로자> 첫 일자리', 'Y - 첫 직장(일자리)에 대한 전반적인 만족도',
       'X3 - 진로선택 및 취업준비 프로그램 1- 진로, 취업 관련 교과목_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 2- 직장체험프로그램(인턴 포함)_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 3- 인적성 검사 등 직업심리검사_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 4- 교내 취업박람회_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 5- 진로 관련 개인 및 집단상담 프로그램_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 6- 면접기술, 이력서 작성 프로그램_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 7- 취업캠프_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 8- 기업채용설명회_참여 여부',
       'X3 - 진로선택 및 취업준비 프로그램 9- 기타_참여 여부', 'X1 - 졸업 평점 점수',
       'X3 - 대학 재학 중 일자리 경험의 유무', 'X3 - 어학연수 경험 횟수', 'X3 - 자격증 소지 개수',
       'X3 - 취업 준비활동 경험여부 1- 기업체 직무적성검사(SSAT 등) 공부',
       'X3 - 취업 준비활동 경험여부 2-영어 등 외국어 공부', 'X3 - 취업 준비활동 경험여부 3- 봉사활동',
       'X3 - 취업 준비활동 경험여부 4-공모전 수상', 'X3 - 취업 준비활동 경험여부 5-자격증 준비',
       'X3 - 취업 준비활동 경

In [191]:
final_columns = [
    "X1 - 성별", "X1 - 연령대",
    "X1 - 졸업 평점 점수", "X1 - 졸업 평점 결측", "X1 - 혼인 여부", 
	"X1 - 부모학력", "X1 - 부모소득구간",
    "X2 - 설립유형_국공립", "X2 - 소재지_지방", "X2 - 전공그룹", "X2 - 학교유형_4년제이상", 
	"X3 - 진로교육_참여개수", "X3 - 취업준비활동_참여개수",
    "X3 - 대학 재학 중 일자리 경험의 유무", "X3 - 어학연수 경험 횟수", "X3 - 자격증 소지 개수",
    "Y - 대학 졸업 후 일자리를 가진 경험의 유무",
    "Y - <임금근로자> 첫 일자리", "Y - <비임금근로자> 첫 일자리", "Y - 첫 직장(일자리)에 대한 전반적인 만족도",
    "Y - 전년도 근로/사업 소득액", "Y - log 소득",
    "T1 - 중소기업 청년취업인턴제_들어본 경험", "T1 - 중소기업 청년취업인턴제_참여경험",
    "T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_들어본 경험",
    "T2 - 행정 인턴제(중앙 행정 기관, 공공 기관 등)_참여경험",
    "T3 - 취업성공패키지_청년고용정책 인지여부", "T3 - 취업성공패키지_청년고용정책 참여 여부",
    "T4 - 내일배움카드(구직업 능력 개발 계좌제)_들어본 경험",
    "T4 - 내일배움카드(구직업 능력 개발 계좌제)_참여경험"
]


final_variable_mapping = [
	"X1_성별", "X1_연령대", "X1_졸업평점점수", "X1_졸업평점결측", "X1_혼인여부", 
	"X1_부모학력", "X1_부모소득구간",
	"X2_설립유형_국공립", "X2_소재지_지방", "X2_전공그룹", "X2_학교유형_4년제이상",
	"X3_진로교육_참여개수", "X3_취업준비활동_참여개수",
    "X3_대학_재학_중_일자리_경험의_유무", "X3_어학연수_경험_횟수", "X3_자격증_소지_개수",
	"Y_대학_졸업_후_일자리를_가진_경험의_유무",
	"Y_임금근로자_첫_일자리", "Y_비임금근로자_첫_일자리", "Y_첫_직장에_대한_전반적인_만족도",
	"Y_전년도_근로_사업_소득액", "Y_log_소득",
	"T1_중소기업_청년취업인턴제_들어본_경험", "T1_중소기업_청년취업인턴제_참여경험",
	"T2_행정_인턴제_들어본_경험", "T2_행정_인턴제_참여경험",
	"T3_취업성공패키지_들어본_경험", "T3_취업성공패키지_참여경험",
	"T4_내일배움카드_들어본_경험", "T4_내일배움카드_참여경험"
]

In [192]:
# 최종 데이터프레임 생성
column_mapping = dict(zip(final_columns, final_variable_mapping))

df_final = df_selected.copy()
df_final = df_final[final_columns]
df_final.rename(columns=column_mapping, inplace=True)

df_final.info()

<class 'pandas.core.frame.DataFrame'>
Index: 17908 entries, 0 to 18270
Data columns (total 30 columns):
 #   Column                    Non-Null Count  Dtype   
---  ------                    --------------  -----   
 0   X1_성별                     17908 non-null  int64   
 1   X1_연령대                    14319 non-null  category
 2   X1_졸업평점점수                 17908 non-null  float64 
 3   X1_졸업평점결측                 17908 non-null  int32   
 4   X1_혼인여부                   17908 non-null  int32   
 5   X1_부모학력                   16705 non-null  float64 
 6   X1_부모소득구간                 12048 non-null  category
 7   X2_설립유형_국공립               17908 non-null  int32   
 8   X2_소재지_지방                 17908 non-null  int32   
 9   X2_전공그룹                   17908 non-null  category
 10  X2_학교유형_4년제이상             17908 non-null  int32   
 11  X3_진로교육_참여개수              17908 non-null  int64   
 12  X3_취업준비활동_참여개수            17908 non-null  int32   
 13  X3_대학_재학_중_일자리_경험의_유무     17908 non-null  int64   


In [194]:
file_path = os.path.join(ROOT, "Data", "df_processed.csv")

df_final.to_csv(file_path, index=False, encoding="utf-8-sig")